akka-actor/src/main/scala/akka/actor/LightArrayRevolverScheduler.scala

```diff
@@ -55,12 +55,17 @@ class LightArrayRevolverScheduler(
   import LightArrayRevolverScheduler._
 
   private val oneNs = Duration.fromNanos(1l)
-  private def roundUp(d: FiniteDuration): FiniteDuration =
-    try {
-      ((d + TickDuration - oneNs) / TickDuration).toLong * TickDuration
-    } catch {
-      case _: IllegalArgumentException ⇒ d // rounding up Long.MaxValue.nanos overflows
+  private def roundUp(d: FiniteDuration): FiniteDuration = {
+    val dn = d.toNanos
+    if (dn <= 0)
+      Duration.Zero
+    else if (dn > Long.MaxValue - tickNanos)
+      d // rounding up Long.MaxValue.nanos overflows
+    else {
+      val r = ((dn - 1) / tickNanos + 1) * tickNanos
+      if (r != dn) r.nanos else d
```

Comment:
Now I think I found it: if this condition is changed to `if (r != dn && r > 0)` then the second case (lines 62–63) can be removed completely.